### DISCUSSION TITLE
Connect Seb-Browser to Seb-Server

### DISCUSSION THREAD

**Discussion by ggschmidt** [SafeExamBrowser/seb-server#41] [Q&A]
Sorry, maybe I am stupid.

I have windows with Safe-Exambrowser
I have a SEB-Server VM running (Version Testing)
I have an LMS and already conducted exams here using Safe-Exambrowser only.

But no idea to connect SEB-Browser and SEB-Server (step 2 of the overview).

(I only want to see if students quit Seb-Browser, deliver seb-Browser config files, students connected to LMS, no proctering)

I can not find this point in the docu or a place in the SEB-Configuration tool.

kind regards
gerd

**Comment by dbuechel** [comment]
You'll need to export the connection configuration from SEB-Server and use that configuration to start SEB. Further information can be found in the user guide for SEB-Server: https://seb-server.readthedocs.io/en/latest/connection_config.html

**Comment by ggschmidt** [comment]
Hello Damian,

 

thanks a lot

I really spent time here (and there) but could not find your link. I am sure it will help.

**Comment by ggschmidt** [comment]
Thanks for help Damian,

 

My intention is to connect seb – sebserver – ILIAS LMS. But without any plugin in Ilias.

We only want to connect SEB with Ilias, and in SEB-Server monitor if SEB was quit/restarted/reconnected.

(addionally allow 3. Party tools).

 

The SEB-User (student) will see only the course and exam he/she is member decided by ILIAS. 

SEB-SERVER has no information on the Exams.

 

Is this possible at all?

 

I am testing SEB on a VM connecting to SEB-Server 1.1. (testing setup).

 

Thanks in advance

 

sorry I did not find a link out of https://seb-server-setup.readthedocs.io/en/latest/overview.html and was lost.

**Reply by dbuechel** [reply]
That looks possible to me, at least from the perspective of the SEB client application. For the server aspect, @anhefti would know better what's possible and meaningful.

**Comment by anhefti** [comment]
Hi! You need to bind your LMS to the SEB Server to "import" the courses or quizzes from the LMS and create an "Exam" on the SEB Server where you can define SEB configurations and more.

Unfortunately ILIAS is not supported yet. There is full support for Open edX, Open OLAT, AnsDelft and partly support for Moodle. If you have one of these LMS you can select the "LMS Setup" section on the SEB Server, "Add new LMS Setup" and then choose one of this LMS types to bind the LMS by providing an API access account from the LMS to the SEB Server.

**Comment by ggschmidt** [comment]
Hello Andreas,

 

Thank you very much for your reply. I tried a lot with ex- and Importing seb configuration but got timeouts by seb-browser.

 

This means: without touching Ilias (have a plugin or something like that), I cannot get a “lightweight” montoring of the seb-browser on the clients (running, restarted, off).

Is that right?

 

Thanks 

Gerd

 

 

Von: Andreas Hefti ***@***.***> 
Gesendet: Montag, 15. November 2021 08:05
An: SafeExamBrowser/seb-server ***@***.***>
Cc: ggschmidt ***@***.***>; Author ***@***.***>
Betreff: Re: [SafeExamBrowser/seb-server] Connect Seb-Browser to Seb-Server (Discussion #41)

 

Hi! You need to bind your LMS to the SEB Server to "import" the courses or quizzes from the LMS and create an "Exam" on the SEB Server where you can define SEB configurations and more.

Unfortunately ILIAS is not supported yet. There is full support for Open edX, Open OLAT, AnsDelft and partly support for Moodle. If you have one of these LMS you can select the "LMS Setup" section on the SEB Server, "Add new LMS Setup" and then choose one of this LMS types to bind the LMS by providing an API access account from the LMS to the SEB Server.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub <https://github.com/SafeExamBrowser/seb-server/discussions/41#discussioncomment-1642671> , or unsubscribe <https://github.com/notifications/unsubscribe-auth/AECGC6VKKR5CYXZIGHSEUD3UMCWIRANCNFSM5HYM6DVQ> .
Triage notifications on the go with GitHub Mobile for iOS <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>  or Android <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub> .   <https://github.com/notifications/beacon/AECGC6R7WYTGMSP53WGBAYLUMCWIRA5CNFSM5HYM6DV2YY3PNVWWK3TUL52HS4DFWFCGS43DOVZXG2LPNZBW63LNMVXHJKTDN5WW2ZLOORPWSZGOAAMRBLY.gif>

**Reply by jemedeiros** [reply]
So there are no direct connections from SEB-Server to SEB-Clients and SEB-Server gets infrmation from SEB-Client when medated by a LMS Setup, right ? No ILIAS API, no monitoring for ILIAS Exams, correct ?

**Reply by anhefti** [reply]
Yes, SEB Server needs a specific LMS integration for secure setup and monitoring of exams with SEB and SEB Server. This is already done for some LMS like OpenOLAT, Moodle or Open edX and some e-assessment suites like ANS. And yes, ILIAS is really missing here.

If you’ve a setup with SEB Server, you can "import" an Exam on SEB Server by using the LMS integration for a specific LMS and configure the respective exam for SEB. SEB Server only needs some key data of a course or quiz from the LMS for that like name, start-,end-time, start URL and some few others. A SEB client than always connects firstly to the SEB Server and gets initialized by the SEB Server for the respective exam before been forwarded to the LMS. During the exam a SEB client has a connection to the SEB Server as well as the usual browser-connection to the LMS. 

An LMS integration consists mainly of two parts, thirst part is the course access API and a second part is the SEB restriction API. The course access API is at least required to integration a specific LMS with SEB Server for SEB monitoring. It allows you to connect to an LMS from SEB Server and request needed course data and user data for the setup and monitoring with SEB Server. 

The integration effort on SEB Server is mainly the implementation of this two API's for a specific LMS. On the LMS site, depending on existing API functionality of the LMS, this needs more or less integration and development effort. The integration effort on SEB client is minimal if needed at all (depends on LMS integration)

> **For example, Open edX integration:**
> 
> Open edX already had an API for the course access part, to request course and user data. So this needed no special integration work on Open edX. The second part, SEB restriction, was not part of the Open edX core LMS and has been implemented by third company partner as a plugin. The integration work on SEB Server and SEB was done "in-house" within the scope of the Swiss MOOC Service project, but there is also the possibility to do this with a third company partner that contributes to the SEB Server open source project for example. 


If you are interested in such an SEB - SEB Server integration for ILIAS we would recommend you to contact [Stephan Kergomard (HS Luzern)](https://www.hslu.ch/de-ch/zllf/lehr-und-lerninfrastruktur/lernplattform-ilias/) and  [Jonas Lanz (UniBas)](https://www.unibas.ch/en/University/Administration-Services/Vice-President-s-Office-for-Education/Learning-and-Teaching/Educational-Technologies/Steckbrief-ADAM.html)  that are both involved in ILIAS integration, to coordinate and finance a cooperated development with an ILIAS development partner.
Furthermore we recommend to consider a cooperated [SEB alliance](https://safeexambrowser.org/alliance/members.html) membership for financing and also to get additional support from SEB developers.

**Reply by jemedeiros** [reply]
Oh wow. Thank you so much for this very complete answer.  I am sincerely grateful.

**Comment by anhefti** [comment]
> This means: without touching Ilias (have a plugin or something like that), I cannot get a “lightweight” montoring of the seb-browser on the clients (running, restarted, off).

Yes this is right. We not only need a plugin or integration on Ilias side but also on the SEB Server side to be able to let SEB Server talk with Ilias. Unfortunately there is no plan at the moment for such an integration for the next release of SEB Server. 

For a more detailed understanding of how SEB Server integrates with an LMS and SEB and how to setup exams and deliver SEB configurations to setup SEB to connect to SEB Server, please have a look at the [SEB Server User Guide](https://seb-server.readthedocs.io/en/latest/#)

**Reply by roetsc** [reply]
Is there a ping web API available that I could run from a python/perl/java script ?  I could configure this script as a startup-script in my browser.

**Reply by anhefti** [reply]
Hi @roetsc 

What do you exactly mean by "ping web API" and for which service/connection?

-  SEB - SEB Server
- SEB - Illias
- SEB Server - Illias

To achieve a proper and secure monitoring with SEB and SEB Server we need an integration on all three parts. 
But there were other requests about Ilias integration lately, so maybe this will be next LMS integration with SEB Server. I will keep you informed about that.

**Reply by roetsc** [reply]
Hi Andreas,
We are currently using Leho, and I am pretty sure there are lots of other LMS system out there, where it will be impossible to integrate them all (-
For this I am looking for a "generic LSM system" than feeds my seb-server, so that I can see who is connected.
I am experimenting with the demoquiz 1 exam that connects to http://lms.mockup.com/api/ that goes to www.pearson.com, if I start my exam from the client though, the connection to (my) sebserver times out.
So is the ping monitoring from the sebserver to the LMS system, or from the LMS system to the sebserver?
In the second phase, I would be looking for a kind of script that send the information from the students pc to the sebserver in stead of from the LMS system. This would be a week workaround for those that use none supported LMS systems, or no LMS system at all.
Maybe I am overlooking something, but I cannot test my exam config from a monitoring perspective if I don't have a "generic LMS client or server".
I hope this answer is clear, or maybe we should have a chat ?

**Reply by anhefti** [reply]
@roetsc 

Thanks for your information, but actually the pings are sent from SEB to SEB Server (not LMS to SEB Server) and SEB connects to SEB Server with a proper Connection Configuration from SEB Server.

To support a proper and more secure SEB - LMS integration is one of the main purposes of SEB Server and therefore the specific LMS integration and binding is principally. We do not support such a workaround you described above and we also do not support a SEB Server to work with no LMS or with no exam-data from a generic LMS since this is an integral part of the SEB Server architecture and its concept.

If you are interested in an official LMS support with SEB and SEB Server for your LMS, please considering to participate on this project by becoming a [SEB Alliance member](https://safeexambrowser.org/alliance/members.html).

**Reply by danschlet** [reply]
The only way how to use an unsupported LMS or exam system that comes to my mind would be to integrate the unsupported system using LTI into a supported system like Moodle. But we never tried that ourselves, you would need to investigate such a solution yourself.
